File: my-solutions/0027-remove-element/solution.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int neq = 0;

        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] == val) {
                nums.erase(nums.begin() + i);
                --i;
            } else {
                ++neq;
            }
        }

        return neq;
    }
};

File: my-solutions/0027-remove-element/solution.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        search, swap, count = 0, len(nums) - 1, 0
        while search < len(nums) and search <= swap:
            if nums[search] == val:
                count += 1
                nums[search], nums[swap] = nums[swap], nums[search]
                swap -= 1
            else:
                search += 1
        return len(nums) - count
